internal/data/data.go

```diff
@@ -0,0 +1,102 @@
+package data
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
+
+	"github.com/DanWlker/remind/internal/config"
+)
+
+// This does not create the file if it doesn't exist
+func GetTodoFromFile(filename string) ([]TodoEntity, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("os.Open: %w", err)
+	}
+	defer f.Close()
+
+	var (
+		items []TodoEntity
+		dec   = yaml.NewDecoder(f)
```

Comment:
Instead of reading the whole file into memory, then YAML-unmarshaling it, stream the data from the file directly into the YAML decoder.